apps/researcher/src/app/[locale]/objects/[id]/slide-over-gallery.tsx

```diff
@@ -83,18 +108,41 @@ interface OpenSeaDragonViewer {
   image: string;
 }
 
-function OpenSeaDragonViewer({image}: OpenSeaDragonViewer) {
+const OpenSeaDragonViewer = ({image}: OpenSeaDragonViewer) => {
+  const [viewer, setViewer] = useState<Viewer | null>(null);
+
   useEffect(() => {
-    OpenSeaDragon({
-      id: 'openSeaDragon',
-      showNavigationControl: false,
-      showNavigator: true,
-      tileSources: {
+    if (image && viewer) {
+      viewer.open({
         type: 'image',
         url: image,
-      },
-    });
-  }, [image]);
+      });
+    }
+  }, [image, viewer]);
+
+  useEffect(() => {
+    const initOpenseadragon = () => {
+      viewer && viewer.destroy();
+      setViewer(
+        OpenSeaDragon({
+          id: 'openSeaDragon',
+          showNavigator: true,
+          zoomInButton: 'zoom-in',
+          zoomOutButton: 'zoom-out',
+          homeButton: 'home',
+          tileSources: {
+            type: 'image',
+            url: image,
+          },
+        })
+      );
+    };
+
+    initOpenseadragon();
+    return () => {
+      viewer && viewer.destroy();
+    };
+  }, []);
 
```

Comment:
Is is possible to resolve this warning from GitHub Actions?

`React Hook useEffect has missing dependencies: 'image' and 'viewer'. Either include them or remove the dependency array. If 'setViewer' needs the current value of 'image', you can also switch to useReducer instead of useState and read 'image' in the reducer`

Reply:
Yes! Sorry, I missed that one.